### DISCUSSION TITLE
[@mantine/prism] Allow consumers to change the Prism Theme

### DISCUSSION THREAD

**Discussion by fbartho** [mantinedev/mantine#1528] [Feature requests]
Current @mantine/prism `<Prism>` component allows you to override light/dark mode, and otherwise you can change how the highlighter works by changing the core color palettes using `<MantineProvider>` but otherwise you can't change which types of symbols / patterns in your code are highlighted, and with which color.

## Proposal

Expose the call-site of `getPrismTheme` so users can provide their own method matching the same signature.
- https://github.com/mantinedev/mantine/blob/97349accc5632aefabb1a86d91a6f4b221e534cf/src/mantine-prism/src/Prism.tsx#L101-L106
- https://github.com/mantinedev/mantine/blob/97349accc5632aefabb1a86d91a6f4b221e534cf/src/mantine-prism/src/prism-theme.ts#L186-L187

## Alternatives Considered
- Do nothing, if users want to change the styling, they have to do one of the following:
  - install their own prism instance or not use `@mantine/prism`
  - provide a new `<MantineProvider` wrapping every multi-line code block.
  - not use `@mantine/prism`
- Update `<MantineProvider` to expose a hook for customizing a `PrismTheme` **Downsides**:
  - requires exposing the `PrismTheme` types in `@mantine/core` which seems like blurring a package boundary.
  - would require more code change in both `@mantine/core` and `@mantine/prism` & probably feels like a breaking change.

**Comment by rtivital** [comment]
I do not think that it will be implemented any time soon, you can use react prim renderer or other highlighting library to setup your own theme, Mantine provides theme based on its theme colors.

**Reply by fbartho** [reply]
@rtivital I'm happy to implement this, and was going to submit a PR to do so if you don't mind?

**Reply by fbartho** [reply]
Implemented here: https://github.com/mantinedev/mantine/pull/1529

- [x] Base functionality
- [x] Docs: `prism.mdx`
- [x] Docs: Storybook

**Reply by fbartho** [reply]
As you can see in the PR, I’m still using Mantine’s Theme Colors.

I’m just providing a configuration point where people can change what tokens the colors apply to, and specifically which of the theme colors apply to which tokens (allowing higher contrast selections, for example)

This feels like a natural & lightweight extension to @mantine/prism, but if there’s a simple alternative to achieve what I’m trying to do, please let me know!

**Reply by fbartho** [reply]
New PR provided here: https://github.com/mantinedev/mantine/pull/1632

**Reply by cliffoo** [reply]
In case anyone stumbles upon this before [pr#1632](https://github.com/mantinedev/mantine/pull/1632) (or another PR of equal function) is merged, a workaround is to wrap your prism component in its own mantine provider. See documentation on nesting mantine providers [here](https://mantine.dev/theming/mantine-provider/#nested-mantineproviders).

**Reply by cliffoo** [reply]
oops I just realized this is mentioned above, oh well
